Lib/test/test_traceback.py

```diff
@@ -4878,6 +4878,28 @@ class PurePythonSuggestionFormattingTests(
     traceback printing in traceback.py.
     """
 
+    def test_exception_punctuation_handling_with_suggestions(self):
```

Comment:
Is it really the place where to put this test? or should it be somewhere else? (sorry but the file is 5k lines long so it's a bit hard to know...)